instrumentation/hystrix-1.4/src/main/java/io/opentelemetry/instrumentation/auto/hystrix/HystrixDecorator.java

```diff
@@ -17,24 +17,80 @@
 package io.opentelemetry.instrumentation.auto.hystrix;
 
 import com.netflix.hystrix.HystrixInvokableInfo;
+import io.opentelemetry.instrumentation.api.FixedSizeCache;
+import io.opentelemetry.instrumentation.api.Functions;
+import io.opentelemetry.instrumentation.api.config.Config;
 import io.opentelemetry.instrumentation.api.decorator.BaseDecorator;
 import io.opentelemetry.trace.Span;
+import java.util.Objects;
 
 public class HystrixDecorator extends BaseDecorator {
   public static final HystrixDecorator DECORATE = new HystrixDecorator();
 
+  private final boolean extraTags;
+
+  private HystrixDecorator() {
+    extraTags = Config.get().isHystrixTagsEnabled();
+  }
+
+  private static final FixedSizeCache<ResourceNameCacheKey, String> RESOURCE_NAME_CACHE =
```

Comment:
Do we expect enough cardinality to avoid just using a normal `ConcurrentHashMap` without fixed size?

Reply:
That's a call you need to make, but "expect" and "what will actually" can be two different things. `FixedSizeCache` degrades to churning rather than unbounded growth in degenerate cases. Attributing this class to me is a little unfair too, it was @bantonsson's work.